src/cmd7.cc

```diff
@@ -3580,24 +3574,6 @@ void do_cmd_unbeliever()
 			do_cmd_unbeliever_antimagic();
 			break;
 		}
-
-		/* Detect Traps */
-	case 2:
-		{
-			s16b skill = get_skill(SKILL_ANTIMAGIC);
-
-			if (skill < 25)
-			{
-				msg_print("You cannot use your detection abilities yet.");
-				break;
-			}
-
-			detect_traps(DEFAULT_RADIUS);
-
-			if (skill >= 35) destroy_doors_touch();
```

Comment:
Ah, here it is. I don't your changes remove destroy_doors_touch(), right?

I think we should keep the "D" anti-magic ability and just rename it "[D]estroy door" or something.